File: android/sync/src/main/kotlin/com/cacompte/sync/Role.kt
package com.cacompte.sync

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Miroir de `Role.swift` — trois rôles possibles dans une partie partagée. `Host` n'est jamais
 * demandé par un pair qui rejoint, seuls `Contributor` et `Observer` transitent sur le fil. */
@Serializable
enum class Role {
    @SerialName("host")
    Host,

    @SerialName("contributor")
    Contributor,

    @SerialName("observer")
    Observer,
}

File: apple/QuiMeneKit/Sources/Sync/Role.swift
/// Doc 09 — trois rôles possibles dans une partie partagée. `.host` n'est jamais demandé par un
/// pair qui rejoint, seuls `.contributor` et `.observer` transitent sur le fil.
public enum Role: String, Codable, Sendable, Equatable {
  case host
  case contributor
  case observer
}
